modules/osm/intersection.js

```diff
@@ -410,18 +410,8 @@ export function osmIntersection(graph, startVertexId, maxDistance) {
                         if (t.id === way.id) {     // match VIA, TO
                             if (v.length === 1 && v[0].type === 'node' && v[0].id === entity.id) {
                                 matchesViaTo = true;    // match VIA node
-
-                            } else {                    // match all VIA ways
-                                var pathVias = [];
-                                for (k = 1; k < currPath.length; k++) {  // k = 1 skips FROM way
-                                    if (currPath[k][0] === 'w') pathVias.push(currPath[k]);
-                                }
-                                var restrictionVias = [];
-                                for (k = 0; k < v.length; k++) {
-                                    if (v[k].type === 'way') restrictionVias.push(v[k].id);
-                                }
-                                var diff = _difference(pathVias, restrictionVias);
-                                matchesViaTo = !diff.length;
+                            } else if (_every(v, function(via) { return currPath.indexOf(via.id) !== -1; })) {
+                                matchesViaTo = true;    // match all VIA ways
```

Comment:
I have to check this part.. I think the reason I changed this was because `v` can be a via node that happens to match somewhere else along the path.   (adding `&& via.type ==='way'` might be ok)

So the stricter test is that we want to match via ways and that they should all exactly match the via ways that the current path follows.

Reply:
ok, I see the purpose now. but it had a bug for the following case: `currPath = ["w?", "n1"]` and `v = [{type:"node", id: "n2"}]`.

I've replaced the check with a more compact implementation in ae9e9fbdd.